src/plugins/intel_cpu/src/utils/shape_inference/shape_inference_ngraph.cpp

```diff
@@ -16,31 +17,44 @@ NgraphShapeInfer::infer(
     std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>> input_values;
 
     input_static_shapes.reserve(input_shapes.size());
-    for (size_t port = 0; port < iranks.size(); port++) {
-        if (iranks[port] == 0) {
-            input_static_shapes.emplace_back();
-        } else {
-            input_static_shapes.emplace_back(input_shapes[port].get());
+     IShapeInferCommon::Result shape_infer_result;
+    if (m_shape_infer->is_implemented_accessor()) {
+        for (size_t port = 0; port < iranks.size(); port++) {
+            if (iranks[port] == 0) {
+                input_static_shapes.emplace_back();
+            } else {
+                input_static_shapes.emplace_back(input_shapes[port].get());
+            }
```

Comment:
do you mean mean below code is also duplicated in `else` branch? because this code frame is in a loop, and in the loop of `else` branch, there is some other code, so I can't remove this duplication, if remove it, there will be two loop.
```
            if (iranks[port] == 0) {
                input_static_shapes.emplace_back();
            } else {
                input_static_shapes.emplace_back(input_shapes[port].get());
            }
```

Reply:
fixed